query/src/main/scala/filodb/query/exec/ExecPlan.scala

```diff
@@ -370,8 +372,13 @@ abstract class NonLeafExecPlan extends ExecPlan {
     parentSpan.mark("create-child-tasks")
     // Create tasks for all results.
     // NOTE: It's really important to preserve the "index" of the child task, as joins depend on it
+    val parallelism: Int = if (parallelChildTasks)
+                              Runtime.getRuntime.availableProcessors()
+                           else
+                              1
+
     val childTasks = Observable.fromIterable(children.zipWithIndex)
-                               .mapAsync(Runtime.getRuntime.availableProcessors()) { case (plan, i) =>
+                               .mapAsync(parallelism) { case (plan, i) =>
                                  dispatchRemotePlan(plan, parentSpan).map((_, i))
```

Comment:
This only controls the parallelism of the dispatching of tasks, not execution of sub-queries.
For this we need to dispatch, wait for results and only then send the next set of sub-queries.

Reply:
@vishramachandran as per my understanding, if there is no parallelism, underlying tasks are inherently chained. In this case dispatching the execplans, get executed sequentially. It won't dispatch next plan until previous one is complete.